internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -112,11 +117,55 @@ func convertSidecar(s map[string]*manifest.SidecarConfig) ([]*template.SidecarOp
 			EntryPoint:   entrypoint,
 			HealthCheck:  convertContainerHealthCheck(config.HealthCheck),
 			Command:      command,
+			PortMappings: sidecarPortMappings,
 		})
 	}
 	return sidecars, nil
 }
 
+// findContainerNameGivenPort searches through exposed ports in the manifest to find the container name that matches the given port.
+func findContainerNameGivenPort(port int, exposedPorts []manifest.ExposedPort) *string {
+	for _, portConfig := range exposedPorts {
+		if portConfig.Port == port {
+			return aws.String(portConfig.ContainerName)
+		}
+	}
+	return nil
+}
+
+func containerPortMappings(workloadName string, port *uint16, targetPort *int, targetContainer *string, sidecars map[string]*manifest.SidecarConfig) (map[string][]*template.PortMapping, error) {
+	exposedPorts, err := manifest.ExposedPorts(workloadName, port, targetPort, targetContainer, sidecars)
+	if err != nil {
+		return nil, err
+	}
+	return convertPortMapping(exposedPorts), nil
+}
+
+func convertPortMapping(exposedPorts []manifest.ExposedPort) map[string][]*template.PortMapping {
+	portMapping := make(map[string][]*template.PortMapping)
+	// Sort the exposed ports so that the order is consistent and the integration test won't be flaky.
+	sort.Slice(exposedPorts, func(i, j int) bool {
+		return exposedPorts[i].Port < exposedPorts[j].Port
+	})
+	for _, exposedPort := range exposedPorts {
+		port := aws.String(strconv.Itoa(exposedPort.Port))
+		if config, ok := portMapping[exposedPort.ContainerName]; ok {
+			portMapping[exposedPort.ContainerName] = append(portMapping[exposedPort.ContainerName], &template.PortMapping{
+				ContainerPort: port,
+				Protocol:      aws.String(exposedPort.Protocol),
+				Name:          exposedPort.ContainerName + aws.StringValue(port),
+			})
+		} else {
+			portMapping[exposedPort.ContainerName] = append(config, &template.PortMapping{
```

Comment:
wut, it doesn't make sense to use `config` in the `else` block no? can't we eliminate the if-else statements:
When we see duplication like this, let's try to find ways to see if it can be eliminated 💭 
In this scenario, we can see both the `if` and `else` statements have the same declaration, that means we can move this declaration outside of it:
```go
for _, in := range exposedPorts {
   port := strconv.Itoa(in.Port)
   out := &template.PortMapping{
      ContainerPort: aws.String(port),
      Protocol: aws.String(in.Protocol),
      Name: in.ContainerName + port,
   }
   portMapping[in.ContainerName] = append(portMapping[in.ContainerName], out) // should work even when nil
}
```

Reply:
As a side-note, we should 🤨 when we see an `else` statement. Else-statements are great if we need the performance benefit but typically we can write the code like this:
```go
value := commonCase()
if condition {
   value = specialCase()
}
// process value...
```
Instead of:
```
if condition {
  value := specialCase()
  // process value...
} else {
  value := commonCase()
  // process value...
}
```

Reply:
I will keep this thing in mind.